### PR TITLE
fix(hermes): document API bearer token lifecycle and retrieval

### LINKED ISSUE #7175: Document Hermes API_SERVER_KEY restart, rebuild, and retrieval lifecycle
## Description

Document the Hermes `API_SERVER_KEY` lifecycle so operators can distinguish ordinary restart stability from expected rebuild rotation.

The field report claimed the token rotates on every gateway start, but current `main` and likely deployed `v0.0.83` do not reproduce that defect:

- `agents/hermes/runtime-config-guard.py:4811-4825` preserves an existing valid 64-character generated key.
- `agents/hermes/start.sh:2660,2683` uses that idempotent helper for root and non-root startup.
- `test/hermes-runtime-api-key.test.ts:447` proves ordinary restart preservation.
- Missing/malformed values are intentionally replaced, and different sandbox homes receive different keys.
- Rebuild replaces the sandbox home; `.hermes/.env` is user-managed rather than durable state, so a rebuilt sandbox mints a new key.
- `nemoclaw <sandbox> gateway-token --quiet` is the sanctioned machine-readable retrieval path and is already implemented/documented, but its lifecycle is not explained.

Expected contract:

| Operation | Key behavior |
|---|---|
| Hermes gateway restart | Preserve |
| Sandbox stop/start | Preserve for the same sandbox filesystem |
| Host OpenShell gateway restart | Preserve |
| Sandbox rebuild/replacement | Generate a new key |
| Missing/malformed key | Generate a new key |
| Different sandbox | Different key |

If a plain restart changes the value without sandbox replacement, that remains a bug and should include before/after sandbox identity plus redacted mint logs.

## Affected Page

`docs/reference/commands.mdx` (`gateway-token`), rebuild documentation, and the planned headless/server deployment guide.

## Issue Type

Missing documentation

## Suggested Fix

- State that the key is generated once per sandbox home and preserved across ordinary gateway/sandbox/host-gateway restarts.
- State that missing/malformed values and rebuild/replacement generate a new key.
- Give the copyable retrieval command:

  ```bash
  TOKEN=$(nemoclaw <sandbox> gateway-token --quiet)
  ```

- Explain that the sandbox must be running, the value is a password, and implementation-owned `.env` files must not be scraped or edited.
- Add a rebuild warning that the Hermes API bearer token will change and point to `gateway-token --quiet`.
- Add a docs-contract test plus rebuild-level rotation/retrieval coverage; preserve the ordinary-restart negative test.

## Acceptance Criteria

- Docs clearly distinguish restart from rebuild.
- Operators have a supported programmatic retrieval path.
- Exact rotation conditions are documented.
- Rebuild output points to retrieval.
- Regression coverage proves ordinary restart preservation and rebuild rotation.

No open duplicate was found. Related closed issues #5249 and #3180 predate the current token retrieval contract.

### PR DESCRIPTION
**Pull request by harjothkhara**
## Summary

Documents the Hermes `API_SERVER_KEY` lifecycle so operators can tell ordinary restart stability apart from expected rebuild rotation, and adds a rebuild-time notice pointing at the supported retrieval command. Before this change the reference described what `gateway-token` prints but not when the value changes, and the headless quickstart told operators to read the token out of the generated Hermes environment — which contradicts the supported retrieval path.

## Related Issue

Fixes #7175

## Changes

- Document the token lifecycle in the `gateway-token` Hermes reference section: generated once per sandbox home, distinct across sandbox homes, preserved across a gateway restart, a sandbox stop and start, and a host OpenShell gateway restart.
- Document the two regeneration cases: rebuilding or replacing the sandbox, and a startup `API_SERVER_KEY` that is missing or is not exactly 64 lowercase hexadecimal characters. Ordinary restarts that change the token without replacing the sandbox stay a bug, and the page says what evidence to collect.
- Document the retrieval contract: the sandbox must be running, the token is a password, and operators use `gateway-token --quiet` instead of reading or editing `.hermes/.env`.
- Fix the contradiction in the headless quickstart, which pointed operators at the generated Hermes environment rather than the supported command.
- Add the rebuild warning to the rebuild guide and the `rebuild` reference section.
- Print a Hermes-only rebuild notice that the API bearer token changed and how to retrieve the replacement. The notice is worded to fire only as guidance for after recovery completes, so it does not read as a success signal on a rebuild that later bails.
- Cover the docs contract with a test scoped to the `gateway-token` heading boundary rather than a whole-file match, so a correct phrase in the wrong section fails.
- Cover the rebuild notice with a Hermes case, an OpenClaw negative case, and the two post-restore failure paths.
- Extend the Hermes snapshot test to prove state restore does not overwrite a replacement home's `API_SERVER_KEY`, which is the structural guarantee behind rotation.
- Rename the touched test titles to behavior-oriented phrasing with the issue reference as a suffix, per the repo test convention.

Scope note: the issue also mentions a planned headless/server deployment guide. That page does not exist yet, so it is out of scope here.

## Type of Change

- [ ] Code change (feature, bug fix, or refactor)
- [x] Code change with doc updates
- [ ] Doc only (prose changes, no code sample modifications)
- [ ] Doc only (includes code sample changes)

## Quality Gates

- [x] Tests added or updated for changed behavior
- [ ] Existing tests cover changed behavior — justification:
- [ ] Tests not applicable — justification:
- [x] Docs updated for user-facing behavior changes
- [ ] Docs not applicable — justification:
- [x] Sensitive paths changed (security, policy, credentials, preflight, onboarding, inference, runner, sandbox, or messaging)
- [ ] Sensitive-path review completed or maintainer-approved waiver recorded — reviewer/approval link/justification:
- [ ] Non-success, skipped, or missing CI check accepted by maintainer — check name, approval link, and follow-up issue:

## Verification

- [x] PR description includes a `Signed-off-by:` line and every commit appears as `Verified` in GitHub
- [x] Normal `pre-commit`, `commit-msg`, and `pre-push` hooks passed, or `npm run check:diff` passed when hooks were skipped or unavailable
- [x] Targeted behavior tests pass for the current change set, or tests are marked not applicable above — command/result: `npx vitest run src/lib/actions/sandbox/rebuild-post-restore-phase.test.ts test/hermes-api-key-lifecycle-docs.test.ts test/snapshot.test.ts` — 3 files, 57 tests passed.
- [ ] Applicable broad gate passed — `npm test` for broad runtime/test-harness changes; `npm run check` for repo-wide validation/coverage changes — command/result:
- [x] Quality Gates section completed with required justifications or waivers
- [x] No secrets, API keys, or credentials committed
- [x] `npm run docs` builds without warnings (doc changes only) — 0 errors. The 2 reported warnings are pre-existing and unrelated to this change: an unauthenticated redirect check that is skipped without `FERN_TOKEN`, and the site-wide light-mode accent contrast ratio.
- [x] Doc pages follow the [style guide](https://github.com/NVIDIA/NemoClaw/blob/main/docs/CONTRIBUTING.md) (doc changes only)
- [ ] New doc pages include SPDX header and frontmatter (new pages only)

### Coverage limitation

Rebuild rotation is proven structurally, not end to end. Two tests together establish it: state restore does not carry `.hermes/.env` into a replacement home, and separate sandbox homes mint distinct keys. No test drives a real rebuild and compares the token before and after. A live-E2E rotation and retrieval check in `test/e2e/live/rebuild-hermes.test.ts` is the natural follow-up; `e2e-live` is opt-in and outside the required gate for code-changing PRs, so it is not included here. Ordinary-restart preservation keeps its existing direct coverage in `test/hermes-runtime-api-key.test.ts`.

---

Signed-off-by: harjoth <harjoth.khara@gmail.com>

## Summary by CodeRabbit

* **Bug Fixes**
  * Hermes “rebuild” now conditionally warns when the Hermes API bearer token changes, using the supported `gateway-token --quiet` command, and suppresses the notice on certain verification/restore failure paths.
  * Restores now preserve Hermes durable-state files while keeping credential-related behavior consistent with the intended semantics during restore.
* **Documentation**
  * Updated Hermes quickstart and command reference to clarify correct bearer-token sourcing, lifecycle/regeneration rules after rebuilds, and proper `--quiet` usage (not environment/dashboard tokens).
* **Tests**
  * Expanded unit and live E2E coverage for token-change notice behavior, documentation text consistency, restore credential/durable-state semantics, and token stability across repeated retrievals.

### CODE REVIEW

File: test/hermes-api-key-lifecycle-docs.test.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    const commands = readHermesGatewayTokenSection();
+
+    expect(commands).toContain("generated once for each sandbox home");
+    expect(commands).toContain("Different sandbox homes receive different tokens");
+    expect(commands).toContain(
+      "gateway restart, sandbox stop and start, and host OpenShell gateway restart",
+    );
+    expect(commands).toContain("rebuild or replace the sandbox");
+    expect(commands).toContain("missing or is not exactly 64 lowercase hexadecimal characters");
+    expect(commands).toContain("while the existing `API_SERVER_KEY` was present and valid");
```
[the remark this answers is not in the document]
Comment (harjothkhara) [reply test/hermes-api-key-lifecycle-docs.test.ts:46]: This assertion is already scoped to the Hermes subsection. `commands` on line 46 is assigned from `readHermesGatewayTokenSection()` on line 37, not the whole file. No change needed.

File: test/e2e/live/rebuild-hermes.test.ts [hunk 2]
```diff
@@ -159,6 +159,34 @@ function expectEqual(actual: string | undefined, expected: string, message: stri
...
+    "bash",
+    [
+      "-lc",
+      [
+        'token="$(nemoclaw "$SANDBOX_NAME" gateway-token --quiet)"',
+        'case "$token" in ""|*[!0-9a-f]*) exit 2 ;; esac',
+        '[ "${#token}" -eq 64 ] || exit 2',
+        "printf '%s' \"$token\" | sha256sum | cut -d' ' -f1",
+      ].join(" && "),
+    ],
```
[the remark this answers is not in the document]
Comment (harjothkhara) [reply test/e2e/live/rebuild-hermes.test.ts]: Fixed in a628e0aa6. Used `JSON.stringify(host.commandPath)` interpolated into the script, matching the existing pattern in the other live tests rather than positional args.

**Comment by wscurran** [comment]
✨ Thanks for the fix. This documents the Hermes API bearer token lifecycle and retrieval path. Ready for maintainer review.

---
**Related open issues:**
- [#7175 Document Hermes API_SERVER_KEY restart, rebuild, and retrieval lifecycle](https://github.com/NVIDIA/NemoClaw/issues/7175)

---
**Related open issues:**
- [#7175 Document Hermes API_SERVER_KEY restart, rebuild, and retrieval lifecycle](https://github.com/NVIDIA/NemoClaw/issues/7175)

**Comment by miyoungc** [comment]
I'll come back and trigger CI after the conflict is resolved.

**Comment by miyoungc** [comment]
The documentation updates look good to me. Since this PR also introduces new E2E test coverage, I’ll defer that portion—and any related CI follow-up—to the development team.

**Review by ericksoa — APPROVED** [review]
Validated the documentation contract, Hermes-only rebuild messaging, credential-boundary coverage, and the merged live-E2E integration on exact head 856e6ad958cca696e44a16016cabbf5b193d0740. Local verification: 59 targeted tests passed, 4 Hermes rebuild support tests passed, docs built with 0 errors, and all commit/pre-push hooks passed. Sensitive-path review complete.
